cmd/pctl/main.go

```diff
@@ -126,42 +126,109 @@ func installCmd() *cli.Command {
 				DefaultText: "profile_subscription.yaml",
 				Usage:       "Filename to use for the generated content.",
 			},
+			&cli.BoolFlag{
+				Name:  "create-pr",
+				Value: false,
+				Usage: "If given, install will create a PR for the modifications it outputs.",
+			},
+			&cli.StringFlag{
+				Name:        "remote",
+				Value:       "origin",
+				DefaultText: "origin",
+				Usage:       "The remote to push the branch to.",
+			},
+			&cli.StringFlag{
+				Name:        "base",
+				Value:       "main",
+				DefaultText: "main",
+				Usage:       "The base branch to open a PR against.",
+			},
+			&cli.StringFlag{
+				Name:  "repo",
+				Value: "",
+				Usage: "The repository to open a pr against. Format is: org/repo-name",
+			},
 		},
 		Action: func(c *cli.Context) error {
-			profilePath, catalogClient, err := parseArgs(c)
-			if err != nil {
-				_ = cli.ShowCommandHelp(c, "install")
+			// Run installation main
+			if err := install(c); err != nil {
 				return err
 			}
+			// Create a pull request if desired
+			if c.Bool("create-pr") {
+				if err := createPullRequest(c); err != nil {
+					return err
+				}
+			}
+			return nil
+		},
+	}
+}
 
-			branch := c.String("branch")
-			subName := c.String("subscription-name")
-			namespace := c.String("namespace")
-			configValues := c.String("config-secret")
-			filename := c.String("out")
+// install runs the install part of the `install` command.
+func install(c *cli.Context) error {
+	profilePath, catalogClient, err := parseArgs(c)
+	if err != nil {
+		_ = cli.ShowCommandHelp(c, "install")
+		return err
+	}
 
-			parts := strings.Split(profilePath, "/")
-			if len(parts) < 2 {
-				_ = cli.ShowCommandHelp(c, "install")
-				return errors.New("both catalog name and profile name must be provided")
-			}
-			catalogName, profileName := parts[0], parts[1]
+	branch := c.String("branch")
+	subName := c.String("subscription-name")
+	namespace := c.String("namespace")
+	configValues := c.String("config-secret")
+	filename := c.String("out")
 
-			fmt.Printf("generating subscription for profile %s/%s:\n\n", catalogName, profileName)
-			w := &writer.FileWriter{Filename: filename}
-			cfg := catalog.InstallConfig{
-				Branch:        branch,
-				CatalogName:   catalogName,
-				CatalogClient: catalogClient,
-				ConfigMap:     configValues,
-				Namespace:     namespace,
-				ProfileName:   profileName,
-				SubName:       subName,
-				Writer:        w,
-			}
-			return catalog.Install(cfg)
-		},
+	parts := strings.Split(profilePath, "/")
+	if len(parts) < 2 {
+		_ = cli.ShowCommandHelp(c, "install")
+		return errors.New("both catalog name and profile name must be provided")
+	}
```

Comment:
```suggestion
	if len(parts) != 2 {
		_ = cli.ShowCommandHelp(c, "install")
		return errors.New("both catalog name and profile name must be provided in the format catalog-name/profile-name)
	}
```

do we support more slashes, e.g. in profile name? Not sure :thinking:

Reply:
I'll create an issue to fix this for the other commands as well and apply in a separate PR.